### ISSUE TITLE
Redis transaction Problem with AOP

### ISSUE THREAD

**Issue by mevivs** [Impetus/kundera#540]
Example

---

```
@Transactional
public void initDatabase() {
    em.createQuery("DELETE FROM Employee").executeUpdate();   // 1
    em.createQuery("DELETE FROM Company").executeUpdate();    // 2
    // insert new data
}
```

---
## Line 1 execute successfully but line 2 failed with following exception:

redis.clients.jedis.exceptions.JedisDataException: ERR DISCARD without MULTI
at redis.clients.jedis.Protocol.processError(Protocol.java:54)
at redis.clients.jedis.Protocol.process(Protocol.java:61)
at redis.clients.jedis.Protocol.read(Protocol.java:122)
at redis.clients.jedis.Connection.getStatusCodeReply(Connection.java:152)
at redis.clients.jedis.BinaryTransaction.discard(BinaryTransaction.java:41)
at com.impetus.client.redis.RedisTransaction.onRollback(RedisTransaction.java:87)
at com.impetus.kundera.persistence.Coordinator.coordinate(Coordinator.java:86)
at com.impetus.kundera.persistence.KunderaEntityTransaction.onTransaction(KunderaEntityTransaction.java:128)
at com.impetus.kundera.persistence.KunderaEntityTransaction.rollback(KunderaEntityTransaction.java:120)
at com.google.inject.persist.jpa.JpaLocalTxnInterceptor.rollbackIfNecessary(JpaLocalTxnInterceptor.java:152)
at com.google.inject.persist.jpa.JpaLocalTxnInterceptor.invoke(JpaLocalTxnInterceptor.java:73)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:606)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)

**Comment by hank-cp** [comment]
HI Vivek,

Seems this issue hasn't been fixed thoroughly. If you run executeUpdate() and em.persist() in a single transaction, an exception will be thrown. And the follow transaction wouldn't be created successfully. See my updated test case.

**Comment by mevivs** [comment]
This seems to be an issue with underlying JedisPool API. If you comment out
<property name="kundera.pool.size.max.active" value="10" />

it should work for you.

Looking into this

-Vivek

**Comment by mevivs** [comment]
Hi,
Added a fix for this issue. Should be fine now. 

-Vivek

**Comment by chhavigangwal** [comment]
Fixed in Kundera-2.11.1
